### PR TITLE
Run examples in CI

### PR DESCRIPTION
**Pull request by billschereriii**
- Update examples to provide a zero exit code on successful run
- Add pytest script to run all examples
- Add make target for `make test-examples`
- Add run of `make test examples` to CI script

### CODE REVIEW

**Comment by billschereriii** [comment]
This PR is blocked by [Add clustered Redis testing to SmartRedis CI](https://github.com/CrayLabs/Poseidon/issues/164).

Everything is working except for the fact that the examples are written for clustered Redis and CI uses non-clustered.

**Review by Spartee — CHANGES_REQUESTED** [review]
These should be in a different testing file (yml file) from ``run_test.yml``. Given that SR already has a whopping 34 checks for each check-in, lets run these once ever merged PR.

Changes:
- Separate workflow file
- run once every PR to develop.

File: .github/workflows/run_one_off.yml
Comment (ben-albrecht) [inline .github/workflows/run_one_off.yml:1]: I recommend a name that reflects the contents or frequency of the test suite (if we expect the contents to change often), e.g. `run_examples`,  `run_post-merge_tests` (follows `run_tests` naming, or `post-merge_tests`

**Review by ben-albrecht — COMMENTED** [review]
CI and Makefile changes LGTM, barring one inline comment

File: examples/test_examples.py
Comment (ben-albrecht) [inline examples/test_examples.py:55]: Why do we need this sleep?

**Review by ben-albrecht — APPROVED** [review]
Rest of the changes LGTM.

File: examples/test_examples.py
Comment (billschereriii) [reply examples/test_examples.py:55]: Honestly, I'm not sure. It was in the other test driver scripts so I kept it there

Comment (ben-albrecht) [reply examples/test_examples.py:55]: It would be nice to understand the rationale and document it as a comment here for our future selves, but that's a nit (not PR-blocking).

File: .github/workflows/run_one_off.yml
Comment (billschereriii) [reply .github/workflows/run_one_off.yml]: renamed to run_post_merge_tests.yml

File: examples/test_examples.py
Comment (billschereriii) [reply examples/test_examples.py:55]: @Spartee Do you happen to know?

Comment (Spartee) [reply examples/test_examples.py:55]: on shared systems sometimes it can take an extra couple seconds to spin up

**Review by Spartee — APPROVED** [review]
Thanks for those changes. this looks good.

**Pull request merged**
